### DISCUSSION TITLE
Support summaries in the headline

### DISCUSSION THREAD

**Discussion by tecunningham** [jotaen/klog#45] [Feature ideas & discussions]
I would love if the KLG format could support free text in the headline (the line where the date goes).

Personally I think it would improve readability, e.g. in the following having day-of-the-week in the headlines makes it easier for me to remember what the context was for the work I did that day:

```
2021-02-28 Sunday
   06:00-7:00  #writing
   08:00-10:00 #writing (pareto frontier)
   12:00-12:40 #implicit preferences

2021-03-01 Monday
   06:00-07:00 #writing
   09:10-11:44 #implicit
   14:20-15:50 #implicit
   16:00-17:00 garden
   21:00-22:00 looking at text file time-trackers
   
2021-03-02 Tuesday
    5:40-07:00 #writing
    9:00-12:20 #implicit -
   12:50-13:10 #implicit jon
   13:30-14:00 talking to eduardo about jobs
   14:46-16:12 #implicit working on foundations
   19:14-20:15 try out Pareto frontier with R
   20:25-21:55 #implicit
```

**Comment by jotaen** [comment]
My initial reasoning to not support summaries in the headline was this:

- The headline can contain a [“should total” property](https://klog.jotaen.net/#should-total) `2018-03-25 (7h30m!)` and I was worried that allowing multiple optional things to appear might lead to confusion or ambiguity.
- The headline is the only place where additional properties or information might go, if ever needed in the future. The lines below are already reserved for free text summaries, so I can’t change anything there without breaking peoples’ existing files. Curiously, the [spec currently allows multiple properties to appear](https://github.com/jotaen/klog/blob/main/Specification.md#properties) in order to account for extensibility, although it’s of theoretical nature at the moment.
- Strictly speaking I think that putting the weekday labels along the dates is a view-concern, not a concern of the data format. So a powerful syntax highlighter could generate these labels on the fly and display them inline. (See mockup below) That way you wouldn’t need to spell out the weekday yourself, and you also could never make a mistake like labelling it with the wrong weekday. Actually having such a syntax highlighter is a different story of course, maybe a potential idea for @vladdeSV to consider for the [VSCode highlighter](https://github.com/vladdeSV/vscode-klog)?

That being said, I still don’t want to rule this out categorically, but unfortunately it’s not super trivial either. Let’s keep this issue open, so that we can refer to it.

<img width="430" alt="Screenshot 2021-03-03 at 20 21 37" src="https://user-images.githubusercontent.com/3618384/109860188-21286e80-7c5e-11eb-9119-7468dd516484.png">

**Comment by jotaen** [comment]
(EDIT: “I still want to rule this out categorically” → “That being said, I still **don’t** want to rule this out categorically”)

**Comment by vladdeSV** [comment]
(Not strictly related to the issue, but I have plans to add some sort of inline labels of sorts to my VS Code syntax highligher. So far I had only come up with the idea of having the total duration for the day, but showing the weekday is a great idea.)
